src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -656,6 +672,12 @@ public override void Execute(ICpu cpu)
             }
             object list = cpu.PopValue();
 
+            // Box strings if necessary to allow them to be indexed
+            if (list is String)
+            {
+                list = new StringValue((String)list);
+            }
+
```

Comment:
I think this sort of conversion needs to be in all things pushed/popped to the stack, rather than here in Get/SetIndex where it will miss a lot of the cases.